numpyro/distributions/transforms.py

```diff
@@ -583,6 +582,53 @@ def log_abs_det_jacobian(self, x, y, intermediates=None):
             )
 
 
+class L1BallTransform(Transform):
+    r"""
+    Transforms a uncontrained real vector :math:`x` into the unit L1 ball.
+    """
+    domain = constraints.real_vector
+    codomain = constraints.l1_ball
+
+    def __call__(self, x):
```

Comment:
Does this have nicer properties than the simpler
```py
norm = jnp.sum(jnp.abs(x), -1, True)
return x * ((1 - np.exp(-norm)) / norm)
```

Reply:
Nice idea. By transforming the norm from R+ to (0, 1), we get a desired transform. Do you think jacobian can be computed easily? It look a bit non-trivial to me. The stick breaking process behaves like shear transform, so jacobian is easily computed.

Reply:
I believe that since the change of measure depends only on the first derivative (the Jacobian), the Jacobian determinant should be simply
'''py
(1 - np.exp(-norm)) / norm
'''

Reply:
yeah, it seems that log Jacobian determinant has [simple formula according to wolfram](https://www.wolframalpha.com/input/?i=log+jacobian+determinant+x+%2F+%7Cx%7C+*+%281+-+e%5E%28-%7Cx%7C%29%29+with+respect+to+x): `-norm`. I'll switch to this which is both faster and more numerical stable.

Reply:
I fail to derive a nice formula for jacobian of x / (1 + norm(x)) transform :(

Reply:
What about `x/sqrt(1+norm(x)**2)` which is smoother?

Reply:
Do you have some tricks to calculate the determinant of jacobian? I can derive the jacobian matrix for a `x * f(norm(x))` transform. But the formula for determinant in the case `len(x) == 2` is complicated to generalize to e.g. `len(x) == 3`.

Reply:
How about this argument:
Let `t(x) = x * f(norm1(x))` be an angle-preserving transformation. Then the Jacobian in polar coordinates is an `f`-scaled identity matrix for directions orthogonal to `x` and in the direction tangent to `x` is
```
d/dradius norm2(x * f(norm1(x)))
  = f(norm1(x)) + norm2(x) * f'(norm1(x)) * d/dradius norm1(x)
  = f(norm1(x)) + norm1(x) * f'(norm1(x))
```
where I have used `d/dradius norm1(x) = norm1(x)/norm2(x)`. Thus the Jacobian determinant is
```
det_J = f(norm1(x)) ** x.shape[-1] * (1 + norm1(x) * f'(norm1(x)) * f(norm1(x)))
```
For example in the case `t(x) = x/(1+norm1(x))` the log determinant is
```
log_abs_det_jacobian
  = -log1p(norm1(x)) * x.shape[-1] + log1p(-norm1(x) / (1 + norm1(x)) ** 3))
```
(I probably have some math errors above, but I believe the orthogonal + tangential d/dradius argument will work)

Reply:
Thanks. I have been thinking about this but failed to follow the math and couldn't make the tests pass. :( Though it seems to be slower (if `x.shape[-1]` is large), the stick-breaking process is the same as what Ola constructed in [his example](https://github.com/pyro-ppl/numpyro/pull/1055/files#diff-c9303fefc2543090c64d22f3114e87b5a612220eb9e8b7431ba9836355cafff5R49). On a 2D plane, I guess to generate a pair (a, b) with |a| + |b| < 1, it is easier to use stick-breaking process than to use norm transform. So for now, I am in favor of keeping the transform as-is.